chainermn/communicators/_base.py

```diff
@@ -146,7 +146,10 @@ def recv(self, source, tag):
         if msgtype.is_tuple:
             msg = []
             for shape in msgtype.shapes:
-                buf = numpy.empty(numpy.prod(shape), dtype=numpy.float32)
+                if shape == ():
+                    buf = numpy.empty(1, dtype=numpy.float32)
+                else:
+                    buf = numpy.empty(numpy.prod(shape), dtype=numpy.float32)
```

Comment:
Case analysis increases cost of maintenance and sometimes causes bugs. How about this?: `buf = numpy.empty(int(numpy.prod(shape)), dtype=numpy.float32) `

Reply:
It seems great!